### PR TITLE
Improve the docker-compose example, as well as verify prev section

### PR DESCRIPTION
**Pull request by marregui**
## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by marregui** [comment]
In relation with 
https://stackoverflow.com/questions/58051469/setting-up-cratedb-4-0-4-cluster

File: docs/deployment/containers/docker.txt [hunk 1]
```diff
@@ -192,26 +208,55 @@ Read about Docker Compose specifics `here <https://docs.docker.com/compose/>`_.
...
+      image: crate
+      container_name: crate01
+      hostname: crate01
+      ports:
+        - 4201:4200
+      net: crate
+      volumes:
+        - /tmp/crate/01:/data
+      command: >
+        crate -Cnetwork.host=_site_
```
Comment (seut) [inline docs/deployment/containers/docker.txt]: also this node should have its seed_hosts and initial_master_nodes configured

Comment (marregui) [reply docs/deployment/containers/docker.txt]: Pointing to itself?, or the other two nodes?. Should each node be aware of the other two?

Comment (seut) [reply docs/deployment/containers/docker.txt]: yes each node should be aware of all other nodes as the order in how they are started may not be deterministic and the configuration should not rely on that

File: docs/deployment/containers/docker.txt [hunk 2]
```diff
@@ -219,6 +219,8 @@ Read about Docker Compose specifics `here <https://docs.docker.com/compose/>`_.
         - /tmp/crate/01:/data
       command: >
         crate -Cnetwork.host=_site_
+        -Cdiscovery.seed_hosts=crate02,crate03
+        -Ccluster.initial_master_nodes=crate02,crate03
```
Comment (seut) [inline docs/deployment/containers/docker.txt]: also the local node should be included in `initial_master_nodes` to also be part of the initial master voting.

Comment (marregui) [reply docs/deployment/containers/docker.txt]: I suppose for all nodes?

Comment (seut) [reply docs/deployment/containers/docker.txt]: yep

File: docs/deployment/containers/docker.txt [hunk 3]
```diff
@@ -95,12 +105,15 @@ user-defined ``crate`` network, so any subsequent nodes can access this node
...
+so::
 
     sh$ docker run -d --name=crate02 \
-          --net=crate -p 4202:4200 --env CRATE_HEAP_SIZE=2g \
+          --net=crate -p 4202:4200 \
+          --env CRATE_HEAP_SIZE=2g \
           crate -Cnetwork.host=_site_ \
-                -Cdiscovery.seed_hosts=crate01
+                -Cdiscovery.seed_hosts=crate01 \
+                -Ccluster.initial_master_nodes=crate01
```
Comment (seut) [inline docs/deployment/containers/docker.txt]: `crate02` should be also added to the `initial_master_nodes`

**Comment by seut** [comment]
@marregui The link in my recent comment was wrong, updated it, please address.

File: docs/deployment/containers/docker.txt [hunk 4]
```diff
@@ -101,40 +113,50 @@ You can visit the admin UI in your browser with this URL:
...
+                -Ccluster.initial_master_nodes=crate01,crate02
+
+Notice here that:
 
-Notice here how we also updated the port mapping, so that port ``4200`` on the
-container is mapped to ``4202`` on our host machine.
+- We also updated the port mapping, so that port ``4200`` on the container
+  is mapped to ``4202`` on our host machine
+- We have set parameter ``discovery.seed_hosts``, which defines the list of
+  master-eligible nodes. By default, if the parameter is not set, local ports
```
Comment (seut) [inline docs/deployment/containers/docker.txt]: "... which defines the list of discovery nodes"

File: docs/deployment/containers/docker.txt [hunk 5]
```diff
@@ -101,40 +113,50 @@ You can visit the admin UI in your browser with this URL:
...
+  in the range 4300 to 4400 will be scanned. Otherwise, the format is a comma
+  separated list of ``host:port`` entries, where port defaults to setting
+  ``transport.tcp.port``.
```
Comment (seut) [inline docs/deployment/containers/docker.txt]: Maybe add explanation of `cluster.initial_master_nodes` here as well?

**Comment by marregui** [comment]
I believe the flow now is correct. At first, a node is created with a self reference for initial master node. So crate01 becomes master on a single node cluster. Then we decide to extend the cluster by adding a second node, using crate01 as seed and both crate01 and crate02 as master voters. 
The way the third node is defined, it cannot vote in a master election. I hope this is correct.

What was a little misguiding for me is the fact that, unless explicitly defined, nodes take a random name. Maybe, as usually we only run a single node per host, nodes should take, by default, the host's name, unless while scanning 4300... another node responds, in which case perhaps subsequent nodes could be called a random name. This way, in docker examples, node and container name would match, without having to define two separate things?

File: docs/deployment/containers/docker.txt [hunk 6]
```diff
@@ -136,10 +136,13 @@ Notice here that:
 - We also updated the port mapping, so that port ``4200`` on the container
   is mapped to ``4202`` on our host machine
 - We have set parameter ``discovery.seed_hosts``, which defines the list of
-  master-eligible nodes. By default, if the parameter is not set, local ports
+  discovery nodes. By default, if the parameter is not set, local ports
   in the range 4300 to 4400 will be scanned. Otherwise, the format is a comma
   separated list of ``host:port`` entries, where port defaults to setting
-  ``transport.tcp.port``.
+  ``transport.tcp.port``
+- ``cluster.initial_master_nodes``: Is not set to both nodes crate01 and
```
Comment (seut) [inline docs/deployment/containers/docker.txt]: Is this a typo and should be "Is now set.." instead?

**Pull request merged**